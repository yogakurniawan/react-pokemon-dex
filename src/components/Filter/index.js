import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'react-flexbox-grid';
import SelectField from 'material-ui/SelectField';
import MenuItem from 'material-ui/MenuItem';
import uuid from 'uuid/v1';

function Filter(props) {
  return (
    <Row start="xs">
      <Col style={{ margin: '0 auto' }} xs={12} sm={10} md={10} lg={6}>
        <SelectField
          style={{ marginLeft: 8 }}
          onChange={props.onChange}
          name="Filter"
          value={props.value}
          label="Filter"
          floatingLabelText="Filter"
        >
          <MenuItem value="all" primaryText="All Types" />
          {
            props.items && props.items.map(item =>
              <MenuItem key={uuid()} value={item.name} primaryText={item.name} />
            )
          }
        </SelectField>
      </Col>
    </Row>
  );
}

Filter.propTypes = {
  onChange: PropTypes.func,
  items: PropTypes.array,
  value: PropTypes.string
};

export default Filter;
