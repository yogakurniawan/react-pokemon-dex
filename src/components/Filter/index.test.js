import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Filter from './index';

injectTapEventPlugin();

const props = {
  items: [],
  value: "",
  onChange: jest.fn()
}

describe('Filter', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <Filter {...props} />
      </MuiThemeProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});