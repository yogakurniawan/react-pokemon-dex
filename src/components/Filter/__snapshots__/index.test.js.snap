// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter renders correctly 1`] = `
<div
  className="row start-xs"
>
  <div
    className="col-xs-12 col-sm-10 col-md-10 col-lg-6"
    style={
      Object {
        "margin": "0 auto",
      }
    }
  >
    <div
      className={undefined}
      label="Filter"
      name="Filter"
      style={
        Object {
          "backgroundColor": "transparent",
          "cursor": "auto",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 16,
          "height": 72,
          "lineHeight": "24px",
          "marginLeft": 8,
          "muiPrepared": true,
          "position": "relative",
          "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
        }
      }
    >
      <label
        className={undefined}
        htmlFor="Filter-undefined-Filter-20877"
        onClick={undefined}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.3)",
            "lineHeight": "22px",
            "muiPrepared": true,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 38,
            "transform": "scale(1) translate(0, 0)",
            "transformOrigin": "left top",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "zIndex": 1,
          }
        }
      >
        Filter
      </label>
      <div
        className={undefined}
        id="Filter-undefined-Filter-20877"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        style={
          Object {
            "WebkitOpacity": 1,
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "rgba(0,0,0,0)",
            "border": "none",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "inherit",
            "display": "block",
            "font": "inherit",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 15,
            "height": "100%",
            "marginTop": 14,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "width": "100%",
          }
        }
      >
        <div
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
              "height": "100%",
              "muiPrepared": true,
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "content": "' '",
                "display": "table",
                "muiPrepared": true,
              }
            }
          />
          <div
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "height": "56px",
                "lineHeight": "56px",
                "muiPrepared": true,
                "opacity": 1,
                "overflow": "hidden",
                "paddingLeft": 0,
                "paddingRight": 56,
                "position": "relative",
                "textOverflow": "ellipsis",
                "top": 6,
                "whiteSpace": "nowrap",
              }
            }
          >
            
          </div>
          <button
            disabled={false}
            href={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOut={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                "background": "none",
                "border": 10,
                "boxSizing": "border-box",
                "cursor": "pointer",
                "display": "inline-block",
                "fill": "#e0e0e0",
                "fontFamily": "Roboto, sans-serif",
                "fontSize": 0,
                "fontWeight": "inherit",
                "height": 48,
                "margin": 0,
                "muiPrepared": true,
                "outline": "none",
                "overflow": "visible",
                "padding": 12,
                "position": "absolute",
                "right": 0,
                "textDecoration": "none",
                "top": 8,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "verticalAlign": null,
                "width": 48,
                "zIndex": 1,
              }
            }
            tabIndex={0}
            type="button"
          >
            <div
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchStart={[Function]}
            >
              <svg
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.87)",
                    "display": "inline-block",
                    "fill": "inherit",
                    "height": 24,
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "userSelect": "none",
                    "width": 24,
                  }
                }
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </button>
          <div
            style={
              Object {
                "borderTop": "none",
                "bottom": 1,
                "left": 0,
                "margin": "-1px 24px",
                "muiPrepared": true,
                "position": "absolute",
                "right": 0,
              }
            }
          />
          <div
            style={
              Object {
                "boxSizing": "border-box",
                "clear": "both",
                "content": "' '",
                "display": "table",
                "muiPrepared": true,
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "display": "none",
            }
          }
        />
      </div>
      <div>
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 1,
              "borderColor": "#e0e0e0",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "width": "100%",
            }
          }
        />
        <hr
          aria-hidden="true"
          style={
            Object {
              "borderBottomStyle": "solid",
              "borderBottomWidth": 2,
              "borderColor": "#00bcd4",
              "borderLeft": "none",
              "borderRight": "none",
              "borderTop": "none",
              "bottom": 8,
              "boxSizing": "content-box",
              "margin": 0,
              "muiPrepared": true,
              "position": "absolute",
              "transform": "scaleX(0)",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;
